servant-server/src/Servant/Server.hs

```diff
@@ -122,13 +124,24 @@ import           Servant.Server.Internal
 -- > main :: IO ()
 -- > main = Network.Wai.Handler.Warp.run 8080 app
 --
-serve :: (HasServer api '[]) => Proxy api -> Server api -> Application
-serve p = serveWithContext p EmptyContext
 
-serveWithContext :: (HasServer api context)
+serve :: (HasServer api '[Bool]) => Proxy api -> Server api -> Application
```

Comment:
I'm against of using `Context` to guide whether to eval or not eval. IMHO single global choice should be enough for now.

Something https://github.com/haskell-servant/servant/issues/309#issuecomment-287222495 is good.

Context won't work for subapis anyway, as we give only single `Context` to `serveWithContext`. And if we use combinators to alter the context, we could be more direct and alter "whether to eval or not".

Reply:
The comment suggests both a global setting & a Context.  Are you requesting both, or just the global?

Reply:
Just global I think.